src/meta/src/hummock/manager/mod.rs

```diff
@@ -3121,27 +3122,6 @@ impl HummockManager {
 
         let mut normal_groups = vec![];
         for cg_id in compaction_group_ids {
-            if versioning.write_limit.contains_key(&cg_id) {
```

Comment:
With these codes removed, you can use  `let compaction_group_ids = self.compaction_group_ids().await;` instead of acquiring versioning guard explicitly to limit the scope of the version read_lock.